test/Tokenizer.t.sol

```diff
@@ -198,22 +210,56 @@ contract TokenizerTest is Test {
         assertEq(tokenContract.balanceOf(bob), 10_000);
     }
 
-    // function testCanUpgradeErc20TokenImplementation() public {
-    //     vm.startPrank(deployer);
-    //     vm.stopPrank();
+    function testCanUpgradeErc20TokenImplementation() public {
+        vm.selectFork(mainnetFork);
 
-    //     vm.startPrank(originalOwner);
-    //     IPToken tokenContractOld = tokenizer.tokenizeIpnft(1, 100_000, "IPT", agreementCid, "");
-    //     vm.stopPrank();
+        address mainnetOwner = 0xCfA0F84660fB33bFd07C369E5491Ab02C449f71B;
+        address mainnetTokenizer = 0x58EB89C69CB389DBef0c130C6296ee271b82f436;
+        address mainnetIPNFT = 0xcaD88677CA87a7815728C72D74B4ff4982d54Fc1;
 
-    //     vm.startPrank(deployer);
-    //     TokenizerNext synthNext = new TokenizerNext();
-    //     tokenizer.upgradeTo(address(synthNext));
-    //     TokenizerNext synth2 = TokenizerNext(address(tokenizer));
+        Tokenizer upgradedTokenizer = Tokenizer(mainnetTokenizer);
+        IPNFT ipnftMainnetInstance = IPNFT(mainnetIPNFT);
+
+        IPToken newIPTokenImplementation = new IPToken();
+        
+        vm.startPrank(mainnetOwner); // Owner address on mainnet
+        tokenizer11 = Tokenizer11(mainnetTokenizer);
+        tokenizer11.upgradeTo(address(new Tokenizer()));
```

Comment:
what's a little dangerous here is that the Tokenizer's IPToken implementation is undefined between these two transactions. I'm wondering if upgradeToAndCall might help (calling the `setIPTokenImplementation` during the upgrade step)